wasp/apps/pager.py

```diff
@@ -82,19 +82,53 @@ class NotificationApp(PagerApp):
 
     def __init__(self):
         super().__init__('')
+        self.confirmation_view = wasp.widgets.ConfirmationView()
+        self.is_in_confirmation_view = False
 
     def foreground(self):
         notes = wasp.system.notifications
-
         id = next(iter(notes))
         note = notes[id]
         del notes[id]
         title = note['title'] if 'title' in note else 'Untitled'
         body = note['body'] if 'body' in note else ''
         self._msg = '{}\n\n{}'.format(title, body)
 
+        wasp.system.request_event(
+            wasp.EventMask.TOUCH
+            | wasp.EventMask.SWIPE_LEFTRIGHT
+        )
         super().foreground()
 
+    def swipe(self, event):
+        mute = wasp.watch.display.mute
+        if event[0] == wasp.EventType.UP:
+            if self._page >= self._numpages:
+                wasp.system.navigate(wasp.EventType.BACK)
+            self._page += 1
+        elif event[0] == wasp.EventType.LEFT:
+            self.confirmation_view.draw('clear notifications?')
+            self.is_in_confirmation_view = True
+        else:
+            if self._page <= 0:
+                wasp.watch.vibrator.pulse()
+                return
+            self._page -= 1
+
+        if not self.is_in_confirmation_view:
```

Comment:
If you move this test into the draw() method then you can use the super-class swipe method to manage up/down scriolling. Then this method ends up more like:

if event[0] == wasp.EventType.DOWN and self._page == 0:
    self.cv.active = True
    mute = wasp.watch.display.mute
    mute(True)
    self._draw()
    mute(False)
else:
    super().swipe(event)

Reply:
Have refactored this to be a bit cleared, thanks for the suggestion

Reply:
> This is a pretty common feature on a lot of smartwatches as someone usually
> will have the notifications loaded onto the watch while they're working and not
> notice they're being added, then check their phone and view them but they'll
> still be present on the watch.

Indeed... but the key point is that this is a bug in the companion app (it failed to clear the notifications from the watch when it was dismissed from the phone) and that fixing that is a better design.

Anyhow... I am happy to merge this since with the additional upward swipe it does fit fairly neatly into the UI. However that doesn't stop it being a workaround for a problem that should not exist ;-) .

One final thing though... the app isn't handling display timeout correctly. If you get the dialog up and then let the watch time out and turn off then the dialog needs to be automatically deactivated (otherwise the next time the user triggers the notification page it goes straight to the confirmation dialog).